keps/sig-scheduling/4328-node-affinity-based-eviction/README.md

```diff
@@ -0,0 +1,426 @@
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP aims to introduce node-affinity-based eviction.
+In this KEP, we will introduce a new nodeAffinity type: `requiredDuringSchedulingRequiredDuringExecution`.
+A new controller called `node-affinity-eviction` will be added to the controller-manager. 
+This controller will monitor changes in node labels and evict pods when their `requiredDuringSchedulingRequiredDuringExecution` node affinity is no longer met.
+
+## Motivation
+
+- Enable `requiredDuringSchedulingRequiredDuringExecution` node affinity for pod scheduling and execution.
+- Provide users with greater control over pod placement.
+- Ensure predictable and consistent pod behavior in various scenarios.
+- Address use cases with specific placement and execution requirements.
+  
+### Goals
+
+- Introduce the `RequiredDuringSchedulingRequiredDuringExecution` **nodeAffinity** type.
+- Add `node-affinity-eviction` controller to ensure pods being evicted if the selectors are no longer met at runtime.
+
+### Non-Goals
+
+- This KEP does not aim to introduce the `RequiredDuringSchedulingRequiredDuringExecution` **podAffinity** type and **podAntiAffinity** type.
+- This feature does not respect PodDisruptionBudget during the eviction.
+
+## Proposal
+- Implement the `requiredDuringSchedulingRequiredDuringExecution` node affinity type.
+- Update the kube-scheduler, kube-controller-manager and kube-apiserver.
+- Define the behavior, syntax, and use cases in documentation.
+- Facilitate testing, adoption, and maintenance of this feature.
+
+### User Stories (Optional)
+
+#### Optimizing Workloads Migrating
+As an engineer of an AI-platform, we have different deployments for inference/training. 
+Our different deployments require different types of gpu.
+For example, some can only run on nodes with `gpu-type-name=T4`, and some can only run on nodes with `gpu-type-name=P100`.
+There are also some deployments which can run on both nodes(They may have made adaptations or they may not need a GPU at all).
+
+Sometimes, our customers want to add some new powerful machines to the cluster.
+For example, some machines with gpu type `T4`, but better ssd and more cpu are added to the cluster.
+As a result, they want to migrate old deployments that must use T4 GPUs to these powerful new machines. 
+As for the old machines, they can continue to run some services that do not require `T4` GPUs.
+
+With this feature, I can simply remove `gpu-type-name=T4` label from the old node and adds it to the new nodes, so I can move target deployments and keep the remaining pods intact.
+
+So far descheduler has the ability to achieve this goal. 
+However, bringing a new component to the system introduces both a lot of uncertainty and additional learning costs. 
+So as a user, I hope that kubernetes can natively support this feature so that users can do it without spending extra learning costs.
+
+#### Manage Multi Tenant Cluster
```

Comment:
```suggestion
#### Manage Multi Tenant Cluster with dedicated nodes
```

Reply:
Also, if this is the primary use case, it should go first.

It could just be the only case, given that the other one is not convincing.

Reply:
I made it the only case, consider the other one is too special.